### PR TITLE
Fix readme render

### LINKED ISSUE #605: README in released 2.2.0 breaks (some?) pip installs
I'm not sure exactly how broad the impact of this is... but note the following build:
https://travis-ci.org/astropy/astropy/jobs/455319988

That seems to be failing due to astropy trying to install asdf 2.2.0, which is then failing due to the non-ascii characters in the README (specifically, the bullets in the "header" of the README)... 

The blunt-force fix would be to just remove the bullets - I'll issue a PR to do just that momentarily (#606).  But if there's some better solution that somehow avoids this reading step I'm all ears...

### PR DESCRIPTION
**Pull request by drdavella**
This is intended to fix #605 and also to fix rendering of the README on pypi. It seems nearly impossible to test without another release, unfortunately.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -89,7 +89,12 @@ install:
     - python setup.py install
 
 script:
-   - $MAIN_CMD $SETUP_CMD
+    - if [[ $TWINE_CHECK ]]; then
+        python setup.py build sdist;
+        twine check dist/*;
```
Comment (eteq) [inline .travis.yml]: Oh, this is diabolical (in a good way)

File: setup_helpers.py [hunk 2]
```diff
@@ -84,9 +85,16 @@ def read_metadata(config_filename):
 
 
 def read_readme(readme_filename):
-    with open(readme_filename) as ff:
+    with codecs.open(readme_filename, encoding='utf8') as ff:
         lines = ff.read().splitlines()
 
+    # Skip lines that contain raw HTML markup
+    lines = lines[:4] + lines[26:]
```
Comment (eteq) [inline setup_helpers.py]: This seems a bit dangerous - what if someone else makes the README and doesn't realize they need to update this file?

Maybe instead look for the `.. raw:: html` marker, and then skip all consecutive following lines that are indented (or empty)?

Comment (drdavella) [reply setup_helpers.py]: Yes, you're absolutely right. We should do something smarter like that in the long term but for the sake of this bug fix I'm going to leave it alone.

Comment (drdavella) [reply setup_helpers.py]: Also the point of the `twine check` above is that it should catch any such changes that lead to invalid output, but we're in agreement.

**Comment by eteq** [comment]
Good points re: the comment @drdavella - I'll make an issue right now to track the "make the html thingie work better", but doesn't have to go in this PR.

File: .travis.yml (hunk 1 above)
Comment (drdavella) [reply .travis.yml]: 😈

**Pull request merged**